File: 프로그래머스/unrated/148653. 마법의 엘리베이터/마법의 엘리베이터.js
let N;

function solution(storey) {
    var answer = 0;
    
    N = storey.toString();
    
    let arr = new Array(N.length);
    
    for (let i = 0 ; i < N.length ; i++) {
        arr[i] = parseInt(N[i]);
    }
    
    for (let i = N.length - 1 ; i > 0 ; i--) {
        if (arr[i] > 5 || (arr[i] === 5 && arr[i - 1] >= 5)) {
            answer += 10 - arr[i];
            arr[i - 1]++;
        } else {
            answer += arr[i];
        }
    }
    
    if (arr[0] > 5) {
        answer += 11 - arr[0];
    } else {
        answer += arr[0];
    }
    
    
    return answer;
}

File: 프로그래머스/unrated/148653. 마법의 엘리베이터/마법의 엘리베이터.java
class Solution {
    public int solution(int storey) {
        int answer = 0;
        
        while (storey != 0) {
            int 몫 = storey / 10;
            int 나머지 = storey % 10;
            if (나머지 < 5 || (나머지 == 5 && 몫 % 10 < 5) ) answer += 나머지;
            else {
                answer += 10 - 나머지;
                몫 += 1;
            }
            
            storey = 몫;
        }
        
        return answer;
    }
}